File: scratch/extract_options.py
import re
import json

with open("scratch/clean.txt", "r", encoding="utf-8") as f:
    text = f.read()

# Find all blocks that look like sub-criteria
# A sub-criteria starts with "CPX.Y"
pattern = re.compile(r'(CP\d+\.\d+\.?.*?)(?=CP\d+\.\d+\.?|$)', re.DOTALL)
matches = pattern.findall(text)

results = {}
for m in matches:
    # Get the CP id
    cp_match = re.search(r'(CP\d+\.\d+)', m)
    if not cp_match: continue
    cp_id = cp_match.group(1)
    
    # Extract the block between Indicator 2 and Indicator 3
    # Indicator 2 ends with "Ya  ☐   Tidak"
    # Indicator 3 starts with "3 "
    block_match = re.search(r'2.*?Ya\s*☐\s*Tidak\s*(.*?)\n\s*3\s+', m, re.DOTALL)
    if block_match:
        block = block_match.group(1)
        # Clean up the block: replace newlines with spaces, fix hyphenation
        clean_block = re.sub(r'\n\s*', ' ', block)
        clean_block = re.sub(r'-\s+', '', clean_block)
        clean_block = re.sub(r'\s+', ' ', clean_block).strip()
        results[cp_id] = clean_block

with open("scratch/extracted_options.json", "w", encoding="utf-8") as f:
    json.dump(results, f, indent=2)

File: scratch/extract_options.js
const fs = require('fs');

const text = fs.readFileSync('scratch/clean.txt', 'utf8');

const pattern = /(CP\d+\.\d+\.?.*?)(?=CP\d+\.\d+\.?|$)/gs;
const matches = text.match(pattern) || [];

const results = {};
for (const m of matches) {
    const cpMatch = m.match(/(CP\d+\.\d+)/);
    if (!cpMatch) continue;
    const cpId = cpMatch[1];
    
    // Extract the block between Indicator 2 and Indicator 3
    const blockMatch = m.match(/2.*?Ya\s*☐\s*Tidak\s*(.*?)\n\s*3\s+/s);
    if (blockMatch) {
        let block = blockMatch[1];
        // Clean up
        block = block.replace(/\n\s*/g, ' ');
        block = block.replace(/-\s+/g, '');
        block = block.replace(/\s+/g, ' ').trim();
        results[cpId] = block;
    }
}

fs.writeFileSync('scratch/extracted_options.json', JSON.stringify(results, null, 2));
